### ISSUE TITLE
Build fails - cannot find React/RCTLog.h

### ISSUE THREAD

**Issue by joelgetaction** [AgoraIO-Community/Agora-RN-Quickstart#6]
Hi,

I've followed the instructions in your README carefully and tried twice, but building in Xcode fails immediately with:
```
/Users/joel/src/getaction/bidvid/Agora-RN-Quickstart/node_modules/react-native/Libraries/Network/RCTNetworkTask.m:8:9: fatal error: 'React/RCTLog.h' file not found
#import <React/RCTLog.h>
        ^~~~~~~~~~~~~~~~
```
Looking at your `ios/Podfile` it looks like you're not including react native or react in your Podfile.

Can somebody help?  I can't get your quickstart to work at all ...

**Comment by Matrixbirds** [comment]
Did you install the cocoapods which write in the ios/Podfile and open `ios/AgoraRNQuickStart.xcworkspace` then do the build steps? I cannot reproduce it, please provide your compiling environment and react-native version here.

**Comment by joelgetaction** [comment]
I did install the cocoapods and follow those those build steps.

I'd be happy to upload a demo repo that you could compile.  It's late at night here - I will do that in the morning and let you know when it's uploaded.  Thanks again.

**Comment by joelgetaction** [comment]
Hello @Matrixbirds I created a fork of this repo at https://github.com/joelgetaction/Agora-RN-Quickstart/branches and made changes on the `joel` branch.  Please clone that forked repo and then `git checkout joel` to be on the `joel` branch.

Then please try to repro the bug using the following exact setup (which is the setup on my machine) so you can be as close to my machine's environment as possible:
* macOS 10.14.3
* nvm 0.34.0
* node v10.15.3 (installed with nvm)
* npm 6.4.1 (installed with nvm and node 10.15.3)
* watchman 4.9.0 (installed via homebrew)
* Homebrew 2.0.3
* bash shell
* iTerm2 build 3.2.8
* Xcode Version 10.1 (10B61)
* react-native-cli 2.0.1
* react-native 0.58.6

I tried compiling both with the new build system and the old legacy build system in Xcode and neither work.  They both produce errors about not being able to locate React header files etc.

I can reproduce the failed build every time on each of several different Macs so it's not happening on just one machine.  I can't get it to build on any of my machines.

Can you confirm that you're able to build it using the setup I described above, including Xcode 10?  Maybe there's something special about your machine like some special links or something that's allowing it to build there.

Either way, thanks for the help, I really appreciate it.  :-). And I am happy to help however I can - just let me know how I can help please.

Thanks @Matrixbirds !

**Comment by Matrixbirds** [comment]
![image](https://user-images.githubusercontent.com/9990676/55129650-d4bf8480-5152-11e9-9765-aab2b50d2016.png)

I just add React dependencies to `Target Dependencies` to fix this problem. I guess the compiler cannot resolve the `React`.

**Comment by joelgetaction** [comment]
Thanks @Matrixbirds .  I tried your fix and it worked!  :-)  I was able to run on an iPhone X.  Let me try again in the morning when my other iPhone is charged up so I can test between two iPhones and make sure the video and audio transmits well.  Then should we update the README to include these dependencies in the build phases?

Also, I'd like to make a test Agora app that uses a more conventional Podfile, like one created from the expo bare-minimum template flow and make sure I can get that to work.  We could possibly include that as a second example app or at least I can verify it in the morning - I'd like to see if I can get a demo app working without needing to add the React dependencies in the Target Dependencies in Xcode.

Let me try that in the morning.

Again, thanks for your help!  I really appreciate your help.  :-)

**Comment by Matrixbirds** [comment]
Thank you for your attention, we are very glad to see your example app. That would be very helpful for every developer.

**Comment by joelgetaction** [comment]
Hi @Matrixbirds I was able to test on 2 iOS devices and the quickstart app now works with the changes you mentioned above.  Also, it built and ran on Android the first time, so I am able to talk between Android-to-iOS and iOS-to-iOS without issue.

I guess we should merge the target dependencies change back into the upstream repo so that others don't have that issue.  Do you want me to create a PR for this?

**Comment by Matrixbirds** [comment]
@joelgetaction sure. Welcome to become contributor.

**Comment by lukewlms** [comment]
For anyone Googling this - adding pods to the Test project worked for me, per this answer: https://github.com/facebook/react-native/issues/12615#issuecomment-321264639
